solidity/contracts/middleware/InterchainAccountRouter.sol

```diff
@@ -307,7 +307,8 @@ contract InterchainAccountRouter is Router {
             _sender,
             _ism.bytes32ToAddress()
         );
-        _interchainAccount.multicall(_calls);
+        // forward all balance to the interchain account for msg.value and non-call value transfers
+        _interchainAccount.multicall{value: address(this).balance}(_calls);
```

Comment:
Did you consider summing the msg.value of the calls and passing that in? This feels kinda blunt, but not sure of the full trade-offs

Reply:
I did not consider that
in what case would we want to leave balance on the router?

Reply:
When composing it with ISMs that pass msg.value non-atomically

Reply:
this solves value arriving non-atomically on the message recipeint (ICA Router) but actually prevents value from arriving to the ICA account itself
it would force all value to flow through this router which I dont think is desirable